test/end2end_test.go

```diff
@@ -1333,6 +1333,47 @@ func testConcurrentServerStopAndGoAway(t *testing.T, e env) {
 	awaitNewConnLogOutput()
 }
 
+func (s) TestDetailedConnectionCloseErrorPropagatesToRpcError(t *testing.T) {
+	rpcDoneOnClient := make(chan struct{})
+	ss := &stubserver.StubServer{
+		FullDuplexCallF: func(stream testpb.TestService_FullDuplexCallServer) error {
+			<-rpcDoneOnClient
+			return status.Error(codes.Internal, "arbitrary status")
+		},
+	}
+	// Use WithBlock() to guarantee that the RPC will be able to pick a healthy connection to
+	// go out on before we call Stop on the server.
```

Comment:
The default "wait for ready" behavior of RPCs should be fine for this, too.  Are you sure this is necessary (and why)?

Reply:
I added this in because NI was still looking for a way to make sure that RPCs failed strictly after the client conn had found a healthy TCP connection for them to go out on, because I was seeing some flakes where the RPC failed with:

```
end2end_test.go:1372: &{0xc0000ba360}.Recv() = _, rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:37309: connect: connection refused", want _, rpc error containing substring: |connection reset by peer| OR |error reading from server: EOF|
```

i.e., in some test flakes, the call to `ss.S.Stop()` caused server shutdown apparently <i>before</i> the RPC picked a connection.

I dealt with this instead by just adding back the `rpcStartedOnServer` channel.

BTW, AFAICS RPCs aren't actually using the `WaitForReady` call option, when using the stub server, right? I do see https://github.com/grpc/grpc-go/blob/950ddd3c37fc38deaf95f3a27b5883af4776a679/internal/stubserver/stubserver.go#L120 in the stub server, but it looks like that basically accomplishes the same thing as `WithBlock`, rather than the `WaitForReady` call option.

Reply:
> BTW, AFAICS RPCs aren't actually using the WaitForReady call option, when using the stub server, right?

Oops.. "wait for ready" is not what I meant - the default is _not_ `WaitForReady`.  But by default, RPCs should wait for the channel to go from connecting->ready (or transient failure, whis is not expected in tests).

I'm not sure why that `waitForReady` call is in the stub server; we should probably remove it and do it in the tests that need it for whatever reason.

But since you are using `ss.Start` which does that `waitForReady` thing, I'm not sure why you'd be getting that RPC error with "connection refused".  That seems like something we should look into.